wheel/python/clvm_rs/program.py

```diff
@@ -0,0 +1,346 @@
+        obj, cursor = cls.from_bytes_with_cursor(
+            blob, 0, calculate_tree_hash=calculate_tree_hash
+        )
+        return obj
+
+    @classmethod
+    def from_bytes_with_cursor(
+        cls, blob: bytes, cursor: int, calculate_tree_hash: bool = True
+    ) -> Tuple[Program, int]:
+        tree = CLVMTree.from_bytes(
+            blob[cursor:], calculate_tree_hash=calculate_tree_hash
+        )
+        obj = cls.wrap(tree)
+        new_cursor = len(bytes(tree)) + cursor
+        return obj, new_cursor
+
+    @classmethod
+    def fromhex(cls, hexstr: str) -> Program:
+        return cls.from_bytes(bytes.fromhex(hexstr))
+
+    def __bytes__(self) -> bytes:
+        if self._cached_serialization is None:
+            self._cached_serialization = sexp_to_bytes(self)
+        if not isinstance(self._cached_serialization, bytes):
+            self._cached_serialization = bytes(self._cached_serialization)
+        return self._cached_serialization
+
+    def __int__(self) -> int:
+        v = self.as_int()
+        if v is None:
+            raise ValueError("can't cast pair to int")
+        return v
+
+    def __hash__(self):
+        return id(self)
+
+    @classmethod
+    def int_from_bytes(cls, b: bytes) -> int:
+        return int_from_bytes(b)
+
+    @classmethod
+    def int_to_bytes(cls, i: int) -> bytes:
+        return int_to_bytes(i)
+
+    # high level casting with `.to`
+
+    def __init__(self):
+        self.atom = b""
+        self._pair = None
+        self._unwrapped = self
+        self._unwrapped_pair = None
+        self._cached_serialization = None
+        self._cached_sha256_treehash = None
+
+    @property
+    def pair(self) -> Optional[Tuple["Program", "Program"]]:
+        if self._pair is None and self.atom is None:
+            pair = self._unwrapped_pair
+            self._pair = (self.wrap(pair[0]), self.wrap(pair[1]))
+        return self._pair
+
+    @classmethod
+    def to(cls, v: Any) -> Program:
+        return cls.wrap(to_clvm_object(v, cls.new_atom, cls.new_pair))
+
+    @classmethod
+    def wrap(cls, v: CLVMStorage) -> Program:
+        if isinstance(v, Program):
+            return v
+        o = cls()
+        o.atom = v.atom
+        o._pair = None
+        o._unwrapped = v
+        o._unwrapped_pair = v.pair
+        o._cached_serialization = getattr(v, "_cached_serialization", None)
+        o._cached_sha256_treehash = getattr(v, "_cached_sha256_treehash", None)
+        return o
+
+    # new object creation on the python heap
+
+    @classmethod
+    def new_atom(cls, v: bytes) -> Program:
+        o = cls()
+        o.atom = bytes(v)
+        o._pair = None
+        o._unwrapped_pair = None
+        return o
+
+    @classmethod
+    def new_pair(cls, left: CLVMStorage, right: CLVMStorage) -> Program:
+        o = cls()
+        o.atom = None
+        o._pair = None
+        o._unwrapped_pair = (left, right)
+        return o
+
+    @classmethod
+    def null(cls) -> Program:
```

Comment:
What do you think about a similar program that uses `1`?  Useful for a lot of testing, and redefining at the top of every file is annoying.

Reply:
I'd be open to that. Do you prefer `.true` or `.one`, or something else?

Reply:
I went with `.one`.